Formalize the overflow and cardinality limit interaction (#3912)
open-telemetry/opentelemetry-specification: specification/metrics/sdk.md

The SDK MUST create an Aggregator with the overflow attribute set
prior to reaching the cardinality limit and use it to aggregate events
for which the correct Aggregator could not be created.  The SDK MUST provide the

> guarantee that overflow does not happen if the maximum number of distinct,

Reviewer: nit:

```suggestion
guarantee that overflow would not happen if the maximum number of distinct,
```

> guarantee that overflow does not happen if the the maximum number of distinct,

Reviewer 2: ```suggestion
guarantee that overflow does not happen if the maximum number of distinct,
```

> non-overflow attributes is less than or equal to the limit.

Reviewer 2: ```suggestion
non-overflow attribute sets is less than or equal to the limit.
```
